src/helper.cc

```diff
@@ -86,18 +86,16 @@ Helper::SessionBase::closePipesSafely(const char * const id_name)
     if (hIpc) {
         if (WaitForSingleObject(hIpc, 5000) != WAIT_OBJECT_0) {
             getCurrentTime();
-            debugs(84, DBG_IMPORTANT, "WARNING: " << id_name <<
+            debugs(84, DBG_IMPORTANT, "WARNING: " << helper().id_name <<
                    " #" << index << " (PID " << (long int)pid << ") didn't exit in 5 seconds");
```

Comment:
I think we can reduce code duplication and improve reporting consistency across similar debugs(), but that should be done in a dedicated minimal PR because those changes will go way beyond HelperServerClosed() deduplication this PR is focusing on. Technically, those changes do not even require helper() introduced by this PR, but they are easier to do now that we have it.